src/parser/types.ts

```diff
@@ -319,6 +319,44 @@ export interface MarkdownComponent extends BaseIndividualComponent {
 
 /**
  * The ReactComponent interface is used to define the properties of a react component. This component is used to render react code with certain parameters. These parameters can be used within your react code to render different things.
+ *
+ * The path is to the location of the react component you want to render, which must be default exported from the designated file.
+ * React components should be placed in src/public/{projectName}, and the path string placed here will be a path from src/public.
+ *
+ * React components created this way have a generic prop type <StimulusParams<T>>, which has the following types.
+ * {
+ *  parameters: T;
+ *  setAnswer: ({ status, provenanceGraph, answers }: { status: boolean, provenanceGraph?: TrrackedProvenance, answers: Record<string, any> }) => void
+ * }
+ *
+ * parameters is the same object passed in from the ReactComponent type.
+ * setAnswer is a callback function allowing the creator of the ReactComponent to programmatically set the answer, as well as the provenance graph. This can be useful if you dont use the default answer interface, and instead have something more unique.
+ * See the click-accuracy-test link below for an example of this.
```

Comment:
This feels a bit confusing to me. On first read, I can't personally understand what this section is describing. I think providing an example of what this is doing would make more sense/provide a better flow. Our goal here is to provide details but making sure these details are rooted in concrete examples. I understand that you're pointing to the 'click-accuracy-test', but I think it would be better to possibly bring in the snippet of code from there. 

I think this example could also be moved to the bottom after you've already given the basics. And maybe lead in with something like "Like our other components, React components come with a default answer interface. However, we do have support for capturing more complex stimuli in React components..."